### PR TITLE
Add scope block support to Data Protection

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**: Add scope block support to Data Protection

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

Tested:

image

image

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

**Review by rnarenpujari — APPROVED** [review]
LGTM. Ideally should move the dir from `internal/resources/cluster/dataprotection` to `internal/resources/dataprotection` since it's no longer specific to cluster. But we can do this later as well if it doesn't cause any issues.

File: docs/resources/enable_data_protection.md [hunk 1]
```diff
@@ -52,6 +54,33 @@ resource "tanzu-mission-control_enable_data_protection" "demo" {
 
 - `id` (String) The ID of this resource.
 
+<a id="nestedblock--scope"></a>
+### Nested Schema for `scope`
+
+Optional:
+
+- `cluster` (Block List, Max: 1) Cluster scope block (see [below for nested schema](#nestedblock--scope--cluster))
+- `cluster_group` (Block List, Max: 1) Cluster group scope block (see [below for nested schema](#nestedblock--scope--cluster_group))
```
[the remark this answers is not in the document]
Comment (shobha2626) [reply docs/resources/enable_data_protection.md:63]: Moving the files/folders around doesn't cause any issues. Only the schema cannot be changed, which we are taking care in this PR

File: docs/resources/enable_data_protection.md [hunk 2]
```diff
@@ -38,9 +42,7 @@ resource "tanzu-mission-control_enable_data_protection" "demo" {
 
 ### Required
 
-- `cluster_name` (String) Cluster name
-- `management_cluster_name` (String) Management cluster name
-- `provisioner_name` (String) Cluster provisioner name
+- `scope` (Block List, Min: 1, Max: 1) Scope block for Data Protection (cluster/cluster group) (see [below for nested schema](#nestedblock--scope))
```
Comment (ramya-bangera) [inline docs/resources/enable_data_protection.md:45]: Do not specify cluster group now. Scope block can accommodate cluster group but do not mention in the schema right now.

File: docs/resources/enable_data_protection.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramya-bangera) [reply docs/resources/enable_data_protection.md:63]: You are right @rnarenpujari we should not be adding it here

**Comment by eastaga** [comment]
LGTM

File: docs/resources/enable_data_protection.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shobha2626) [reply docs/resources/enable_data_protection.md]: Done

File: docs/resources/enable_data_protection.md (hunk 2 above)
Comment (shobha2626) [reply docs/resources/enable_data_protection.md]: Done

**Review by rnarenpujari — APPROVED** [review]

**Review by ankitsny — DISMISSED** [review]

**Review by vmw-vjn — APPROVED** [review]

**Review by ekoop-terasky — APPROVED** [review]

**Review by ankitsny — APPROVED** [review]

**Review by ekoop-terasky — APPROVED** [review]

**Review by rnarenpujari — APPROVED** [review]

**Pull request merged**